src/core/utils/Ledger.ts

```diff
@@ -0,0 +1,238 @@
+     * @param signerPublicKey the public key of signer
+     * @return a signed Transaction which is signed by account at path on Ledger
+     */
+    public async signTransaction(path: string, transferTransaction: any, networkGenerationHash: string, signerPublicKey: string) {
+        const rawPayload = transferTransaction.serialize();
+        const signingBytes = networkGenerationHash + rawPayload.slice(216);
+        const rawTx = Buffer.from(signingBytes, 'hex');
+        const response = await this.ledgerMessageHandler(path, rawTx);
+        // Response from Ledger
+        const h = response.toString('hex');
+        const signature = h.slice(0, 128);
+        const payload = rawPayload.slice(0, 16) + signature + signerPublicKey + rawPayload.slice(16 + 128 + 64, rawPayload.length);
+        const generationHashBytes = Array.from(Convert.hexToUint8(networkGenerationHash));
+        const transactionHash = Transaction.createTransactionHash(payload, generationHashBytes);
+        const signedTransaction = new SignedTransaction(
+            payload,
+            transactionHash,
+            signerPublicKey,
+            transferTransaction.type,
+            transferTransaction.networkType,
+        );
+        return signedTransaction;
+    }
+
+    /**
+     * sign a Symbol Cosignature transaction with a given BIP 44 path
+     *
+     * @param path a path in BIP 44 format
+     * @param transferTransaction a transfer transaction needs to be signed
+     * @param signerPublicKey the public key of signer
+     * @return a Signed Cosignature Transaction
+     */
+    public async signCosignatureTransaction(path: string, cosignatureTransaction: AggregateTransaction, signerPublicKey: string) {
+        const rawPayload = cosignatureTransaction.serialize();
+        const signingBytes = cosignatureTransaction.transactionInfo.hash + rawPayload.slice(216);
+        const rawTx = Buffer.from(signingBytes, 'hex');
+        const response = await this.ledgerMessageHandler(path, rawTx);
+        // Response from Ledger
+        const h = response.toString('hex');
+        const signature = h.slice(0, 128);
+        const cosignatureSignedTransaction = new CosignatureSignedTransaction(
+            cosignatureTransaction.transactionInfo.hash,
+            signature,
+            signerPublicKey,
+        );
+        return cosignatureSignedTransaction;
+    }
+    /**
+     * handle sending and receiving packages between Ledger and Wallet
+     * @param path a path in BIP 44 format
+     * @param rawTx a raw payload transaction hex string
+     * @returns respond package from Ledger
+     */
+    private async ledgerMessageHandler(path: string, rawTx: Buffer) {
+        const TX_INS_FIELD = 0x04;
+        const MAX_CHUNK_SIZE = 255;
+        const CONTINUE_SENDING = '0x9000';
+
+        const chainCode = false;
+        const ed25519 = true;
+
+        const curveMask = ed25519 ? 0x80 : 0x40;
```

Comment:
@TriPhan96 Do we need the conditional operator here while knowing that it will always evaluate to true?

Reply:
Updated to use `0x80` const all the time.